libnmstate/nm/bridge.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import six
+
+from libnmstate.nm import connection
+from libnmstate.nm import nmclient
+from libnmstate.schema import LinuxBridge as LB
+
+
+BRIDGE_TYPE = LB.TYPE
+
+
+def create_setting(iface_state, base_con_profile):
+    bridge = iface_state.get(LB.CONFIG_SUBTREE)
+    if not bridge:
+        return None
+
+    bridge_setting = None
+    if base_con_profile:
+        bridge_setting = base_con_profile.get_setting_bridge()
+        if bridge_setting:
+            bridge_setting = bridge_setting.duplicate()
+
+    if not bridge_setting:
+        bridge_setting = nmclient.NM.SettingBridge.new()
+
+    for key, val in six.viewitems(bridge):
+        if key == LB.MAC_AGEING_TIME:
+            bridge_setting.props.ageing_time = val
+        elif key == LB.GROUP_FORWARD_MASK:
+            bridge_setting.props.group_forward_mask = val
+        elif key == LB.MULTICAST_SNOOPING:
+            bridge_setting.props.multicast_snooping = val
+        elif key == LB.STP_SUBTREE:
```

Comment:
It seems that this subtree handling could belong in its own helper method.

Reply:
Thank you for reminding my own rule: Make the functions as small as possible.